### DISCUSSION TITLE
Writing a tutorial for implementing a general circuit for the U_a gates in Shor's algorithm.

### DISCUSSION THREAD

**Discussion by maxeppler** [Qiskit/.github#34] [Ideas]
Hey guys, 

sorry if this is the wrong channel, I was bored and I've build the Circuit of this paper in Qiskit: https://arxiv.org/abs/quant-ph/0205095 
for a general implementation of the U_a Gates in Shors algorithm. 

When I looked through the Qiskit documentation and the tutorials it seemed that there was once a tutorial for Shors algorithm, but it has been removed. Does anyone know why?

I would offer to write a tutorial on implementing Shors algorithm with general U_a gates since I've basically completed the programming and would just need to explain what I did. 
What does the community think about a tutorial for this circuit? 
Shors algorithm was the first big circuit which I learned in Quantum Computing and I think it might be useful to have a tutorial for a general implementation.

Best regards

**Comment by 1ucian0** [comment]
You can suggest tutorials in https://github.com/Qiskit/documentation/issues/new/choose
